### ISSUE TITLE
[Discussion] Best-practice Documentation

### ISSUE THREAD

**Issue by raedwulf** [sustrik/libdill#58]
Creating this issue as a reminder.

As there's been a lot of discussion relating to the design of applications using libdill, I think we should, eventually, have some FAQ page containing a distilled version on how to use libdill.

**Comment by pskocik** [comment]
The current docs could use some editing too. There's quite a lot of grammar errors plus some stylistic issues in them.

 I've been going through [http://libdill.org/tutorial.html](http://libdill.org/tutorial.html)  and the code is even a little out of sync with the text (E.g., invalid port numbers are expected to elicit an error message, but the way the code is written, the `assert()` after `ipaddr_local()` will kill the program before the `perror()` line is reached). 

I could submit some patches and make some of the docs a little more comfortable to read. Would this be welcome? If so, the HTML (the code examples at least) looks generated. If it is, the original code would help.

**Comment by sustrik** [comment]
Yes. It would be welcome. The text is generated from .md files in gh-pages branch.
